docs/architecture/adr-064-batch-verification.md

```diff
@@ -0,0 +1,82 @@
+# ADR 064: Batch Verification
+
+## Changelog
+
+- January 28, 2021: Created (@marbar3778)
+
+## Context
+
+Tendermint uses public private key cryptography for validator signing. When a block is proposed and voted on validators sign a message representing acceptance or rejection of a block. These signatures are also used to verify previous blocks are correct if a node is syncing. Currently, Tendermint requires each signature to be verified individually, this leads to a slow down of block times.
+
+## Alternative Approaches
+
+- Signature aggregation
+  - Signature aggregation is an alternative to batch verification. Signature aggregation leads to fast verification and smaller block sizes. At the time of writing this ADR there is on going work to enable signature aggregation in Tendermint. The reason why we have opted to not introduce it at this time is because it will slow down verification times. One of the two benefits can be done, aggregation, but verification would suffer because the software would need to verify each signature individually due to their uniqueness, then aggregate them. For example if we were to implement signature aggregation with BLS, there could be a potential slow down of 10x-100x in verification speeds.
```

Comment:
You say first that it leads to faster verification and then say that we will not introduce if because it will slow down verification times. Is this because it's easy to verify it once it has been aggregated but will take a while to actually aggregate all the votes.

Reply:
On a slightly different thread, given the pace of research, do you think it's likely that in the future it may be more beneficial to jump across to signature aggregation?

Reply:
Will reword. The process of signature aggregation can be fast if you can aggregate then verify. But because our messages are unique (different timestamps) we can not aggregate before verification. Verifying a single key with BLS is 10 times slower than a single ed25519 signature verification. We will make the jump to signature aggregation in the future, this is meant to be a speed for projects that dont want to use BLS in the future and who are using ed25519 now.

Reply:
So if we implement BLS in the future, it will be something optional?

Reply:
It would be ideal. BLS is a nice thing for some teams but some of them want to use ed25519 or something else. This conversation will come up only after proposer based timestamps and various other things. Not to concerned with it now.